scrapy_poet/injection.py

```diff
@@ -338,6 +338,10 @@ def is_callback_requiring_scrapy_response(
         # There's no type annotation, so we're probably using response here.
         return True
 
+    if not isinstance(first_parameter.annotation, type):
+        # The annotation is using something like Union, so we're probably using response here.
```

Comment:
It took me some time to underdtand the comment, and the logic. I think the point is that the only case when we should skip the response is when the first argument is annotated exactly as DumyResponse. The mention of Union confused me a lot :) Would it make sense to use https://github.com/scrapinghub/andi/blob/c5a6daae013deabf5c215e87ac1fe671cad665dd/andi/typeutils.py#L28?

Reply:
"like issubclass, but return False if cls is not a class, instead of raising an error" is exactly what we need here :)